pkg/cover/canonicalizer.go

```diff
@@ -59,7 +60,12 @@ func (can *Canonicalizer) NewInstance(modules []host.KernelModule) *Canonicalize
 	instToCanonicalMap := make(map[uint32]*canonicalizerModule)
 	canonicalToInstMap := make(map[uint32]*canonicalizerModule)
 	for _, module := range modules {
-		canonicalAddr := can.modules[module.Name]
+		canonicalAddr, found := can.modules[module.Name]
+		if !found {
+			log.Errorf("instance's modules differ from canonical. Disabling canonicalization")
```

Comment:
Nit: it might help to also include the module name to the log message.

Reply:
Done!